spec/helpers/finance_helper_spec.rb

```diff
@@ -75,6 +75,68 @@
     end
   end
 
+  describe "#voidable_declaration?" do
+    subject { helper }
+
+    ParticipantDeclaration::VOIDABLE_STATES.each do |state|
+      context "when the declaration is #{state}" do
+        let(:declaration) { create(:ect_participant_declaration, state) }
+
+        it { is_expected.to be_voidable_declaration(declaration) }
+      end
+    end
+
+    (ParticipantDeclaration.states.keys - ParticipantDeclaration::VOIDABLE_STATES).each do |state|
+      context "when the declaration is #{state}" do
+        let(:declaration) { create(:ect_participant_declaration, state) }
+
+        it { is_expected.not_to be_voidable_declaration(declaration) }
+      end
+    end
+
+    context "when the declaration is paid and there is a next output fee statement" do
```

Comment:
can we add a test when there is no output fee statement but it's paid?

Reply:
That should be covered by the test above: `(ParticipantDeclaration.states.keys - ParticipantDeclaration::VOIDABLE_STATES)`